src/main/scala/com/bazaarvoice/sswf/model/history/HistoryFactory.scala

```diff
@@ -118,14 +122,23 @@ object HistoryFactory {
             completedStepCounts = completedStepCounts + (scheduledStep.step -> (completedStepCounts(scheduledStep.step) + 1))
 
             assert(scheduledStep == currentStep, s"id[$scheduledStep] != currentStep[$currentStep]")
-            Some(StepEvent[StepEnum](eventId, h.getEventId, Left(scheduledStep), StepResult.serialize(result), eventStart, Some(endTime), new Duration(currentStepStart, endTime)))
+
+            val totalActivityTime: Duration = new Duration(currentStepStart, endTime)
+
+            if (!result.isInProgress) {
+              currentStep = null
+              currentStepStart = null
+            }
```

Comment:
looks like you figured out how to correctly count invocations and cumulative time when the workflow contains consecutive invocations of the same step. Nice work!